app/src/main/java/com/gdsctsec/smartt/viewmodel/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package com.gdsctsec.smartt.viewmodel
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import com.gdsctsec.smartt.data.TimeTable
+import com.gdsctsec.smartt.data.Weekday
+import com.gdsctsec.smartt.data.repository.LectureRepository
+import java.util.*
+
+class HomeScreenViewModel(context: Context) : ViewModel() {
+
+    private lateinit var lecturesOfTheDayLiveList: LiveData<List<TimeTable>>
+
+    init {
+        lecturesOfTheDayLiveList =
+            LectureRepository(context, getWeekday()).getLecturesByWeekday(getWeekday())
```

Comment:
please don't directly instantiate, create a top level reference/variable(like you have done `lecturesOfTheDayLiveList`) for the repository and then use that everywhere